### ISSUE TITLE
Question about Model Loss Flow during second train (Distillation Loss)

### ISSUE THREAD

**Issue by gjustin40** [jiankang1991/TGRS_DisOptNet#4]
I have read your paper and would like to respectfully inquire whether my understanding of it is correct.
I have some questions that I would like to ask for clarification about Distillation Loss

I hope you don't mind me asking, but from my understanding of the formula and code presented in the paper, it seems that only the SAR-Optical Shared Encoder is trained using the distillation loss during the second training. Additionally, it appears that the distillation loss is calculated solely based on the results obtained from the Optical Stage(pre-trained Optical network and SAR train network) and not from the SAR Stage. Could you please confirm if my understanding is correct?

If my understanding is correct, would this mean that the SAR Stage is not affected by the Distillation Loss during backpropagation?
(And I assume you did this because the features of SAR and Optical networks become increasingly dissimilar at higher levels.)

I have attached the code for your reference as well in your code.
`dist_rgb_loss = sum([self.dist_f_loss(sar_f, rgb_f) for sar_f, rgb_f in zip(sar_features, rgb_features)])`
sar_features : Optical Stage features from second Training Process (SAR+RGB training process)
rgb_featuers : Optical Stage features from pre-trained Optical Network

Thanks

**Comment by jiankang1991** [comment]
@gjustin40 Besides the distillation loss, there are also other losses used. You can find the codes:

```
self.seg_loss = get_loss(self.config['training'].get('loss'),
                             self.config['training'].get('loss_weights'),
                             custom_losses)
        
self.dist_f_loss = torch.nn.MSELoss()
        
self.dist_pix_loss = CriterionPixelWise()

```

For distillation, I just force the early stages of encoders to be involved. And the SAR branch is used for capturing the specific SAR semantics.

**Comment by gjustin40** [comment]
I just wanted to confirm whether it's true that the Distillation Loss is not applied to the SAR Stage (4, 5) Encoder. Your answer has resolved my question. 
Thank you! @jiankang1991
